### ISSUE TITLE
withData does not take Nullable types

### ISSUE THREAD

**Issue by Hydragyrum** [kotest/kotest#2718]
I have a use case where I with to pass a `List<String?>` in the `withData` function. However, the function requires a non-nullable type 
`TypeVariable<T>`

eg:

```kotlin
val listOfData = listOf(
  "one",
  "two",
  null
)
withData<String?>({ "Test with $it"}, listOfData) { string ->
  // Test code here
}
```

**Comment by Hydragyrum** [comment]
A workaround is to create a holder class which takes a Nullable type, but it's somewhat awkward.

For example:

```kotlin
@JvmInline
value class NullableStringHolder(val value: String?) {
  override fun toString(): String {
    return "[$value]"
  }
}
```

and use that 

```kotlin
val listOfData = listOf(
  NullableStringHolder("one"),
  NullableStringHolder("two"),
  NullableStringHolder(null)
)
withData<NullableStringHolder>({ "Test with $it"}, listOfData) { string ->
  // Test code here
}
```

This creates a lot of unnecessary boilerplate

**Comment by sksamuel** [comment]
Fixed in 5.0.2
